src/test/socket/bind/test_bind.rs

```diff
@@ -0,0 +1,588 @@
+            for (test_name, _) in more_tests.iter_mut() {
+                *test_name = format!("{} <type={},flag={}>", test_name, sock_type, flag);
+            }
+
+            tests.extend(more_tests);
+        }
+    }
+
+    let num_tests = tests.len();
+    let tests: std::collections::BTreeMap<_, _> = tests.into_iter().collect();
+
+    // make sure we didn't have any duplicate tests
+    assert_eq!(num_tests, tests.len());
+
+    tests
+}
+
+fn get_all_tests() -> std::collections::BTreeMap<String, TestFn> {
+    #[rustfmt::skip]
+    let mut tests: Vec<(String, TestFn)> = vec![
+        ("test_non_socket_fd".to_string(),
+            Box::new(test_non_socket_fd)),
+        ("test_null_addr".to_string(),
+            Box::new(test_null_addr)),
+    ];
+
+    // tests to repeat for different socket options
+    for &sock_type in [libc::SOCK_STREAM, libc::SOCK_DGRAM].iter() {
+        for &flag in [0, libc::SOCK_NONBLOCK, libc::SOCK_CLOEXEC].iter() {
+            #[rustfmt::skip]
+            let mut more_tests: Vec<(String, TestFn)> = vec![
+                // Docker does not support IPv6, so this test is disabled for now
+                //("test_ipv6".to_string(),
+                //    Box::new(move || test_ipv6(sock_type, flag)))
+            ];
+
+            // add details to the test names to avoid duplicates
+            for (test_name, _) in more_tests.iter_mut() {
+                *test_name = format!("{} <type={},flag={}>", test_name, sock_type, flag);
+            }
+
+            tests.extend(more_tests);
+        }
+    }
+
+    let num_tests = tests.len();
+    let mut tests: std::collections::BTreeMap<_, _> = tests.into_iter().collect();
+
+    // make sure we didn't have any duplicate tests
+    assert_eq!(num_tests, tests.len());
+
+    // add all of the passing tests
+    tests.extend(get_passing_tests());
+
+    tests
+}
+
+fn run_tests<'a, I>(tests: I, summarize: bool) -> Result<(), ()>
+where
+    I: Iterator<Item = (&'a String, &'a TestFn)>,
+{
+    for (test_name, test_fn) in tests {
+        print!("Testing {}...", test_name);
+
+        match test_fn() {
+            Err(msg) => {
+                println!(" ✗ ({})", msg);
+                if !summarize {
+                    return Err(());
+                }
+            }
+            Ok(_) => {
+                println!(" ✓");
+            }
+        }
+    }
+
+    Ok(())
+}
+
+// test binding using an argument that cannot be a fd
+fn test_invalid_fd() -> Result<(), String> {
+    let args = BindArguments {
+        fd: -1,
+        addr: None,
+        addr_len: Some(5),
+    };
+
+    check_bind_call(&args, true, Some(libc::EBADF))?;
+
+    Ok(())
+}
+
+// test binding using an argument that could be a fd, but is not
+fn test_non_existent_fd() -> Result<(), String> {
+    let args = BindArguments {
+        fd: 8934,
+        addr: None,
+        addr_len: Some(5),
+    };
+
+    check_bind_call(&args, true, Some(libc::EBADF))?;
```

Comment:
For most/all of these: isn't using the `?` operator here followed by returning `Ok(())` equivalent to just returning this result?  e.g.
```
  ...
  check_bind_call(&args, true, Some(libc::EBADF))
}
```

Reply:
Good catch! I missed that.